File: app/src/main/java/tpcreative/co/qrscanner/common/GenerateSingleton.java
package tpcreative.co.qrscanner.common;
public class GenerateSingleton {
    private static GenerateSingleton instance ;
    private SingletonGenerateListener listener;

    public static GenerateSingleton getInstance(){
        if (instance==null){
            synchronized (GenerateSingleton.class){
                if (instance==null){
                    instance = new GenerateSingleton();
                }
            }
        }
        return instance;
    }
    public void setListener(SingletonGenerateListener listener){
        this.listener = listener;
    }

    public void onCompletedGenerate(){
        if (listener!=null){
            listener.onCompletedGenerate();
        }
    }
    public interface SingletonGenerateListener{
        void onCompletedGenerate();
    }
}

File: app/src/main/java/tpcreative/co/qrscanner/common/GenerateSingleton.kt
package tpcreative.co.qrscanner.common

class GenerateSingleton {
    private var listener: SingletonGenerateListener? = null
    fun setListener(listener: SingletonGenerateListener?) {
        this.listener = listener
    }

    fun onCompletedGenerate() {
        if (listener != null) {
            listener?.onCompletedGenerate()
        }
    }

    interface SingletonGenerateListener {
        fun onCompletedGenerate()
    }

    companion object {
        private var instance: GenerateSingleton? = null
        fun getInstance(): GenerateSingleton? {
            if (instance == null) {
                synchronized(GenerateSingleton::class.java) {
                    if (instance == null) {
                        instance = GenerateSingleton()
                    }
                }
            }
            return instance
        }
    }
}